.github/workflows/main.yml

```diff
@@ -0,0 +1,48 @@
+name: CI Build
+on: [push, pull_request]
+
+jobs:
+  linux:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      
+    - name: Download latest NuGet 
+      run: wget https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
+
+    - name: Install Dependencies
+      run: mono nuget.exe restore
+    
+    - name: Build Project
+      run: xbuild TLSharp.Core/TLSharp.Core.csproj
+      
+  macos:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+      
+    - name: Download latest NuGet
+      run:  curl -o nuget.exe https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
+
+    - name: Install Dependencies
+      run: mono nuget.exe restore
+      
+    - name: Build Project
+      run: MSBuild TLSharp.Core/TLSharp.Core.csproj
+
+  windows:
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    
+    - name: Setup Nuget.exe
+      uses: warrenbuckley/Setup-Nuget@v1
```

Comment:
ok, now I believe Windows doesn't have wget or curl by default, but you can use powershell to download a file, see: https://stackoverflow.com/a/51225744/544947

Reply:
https://github.com/aarani/TLSharp/runs/554107473?check_suite_focus=true

Reply:
i think it's saving the file somewhere else

Reply:
it's failing

Reply:
Fixed

Reply:
Fixed